### ISSUE TITLE
Unclear if dependencies are included

### ISSUE THREAD

**Issue by pilhuhn** [quarkusio/code.quarkus.io#740]
**Describe the bug**

I was looking for Drools CEP on Quarkus and typed in 'drools'  in the filter field, which resulted in 5 hits.
I selected CEP. Now the others did not get marked and it is/was unclear to me if dependencies are
automatically added or not. 

**Expected behavior**

Needed dependencies are either automatically selected. Or some help text says that they will
magically appear in the pom.

**Actual behavior**

User feels unsure what will happen.

**Screenshots**

![Bildschirmfoto 2022-01-19 um 11 16 31](https://user-images.githubusercontent.com/208246/150111272-ebda9624-386f-4a2f-9d46-e83c2979a965.png)

**Environment (please complete the following information):**
 - Browser: 
 - OS: 

**Additional context**
(Add any other context about the problem here.)

**Comment by pilhuhn** [comment]
So apparently only selecting that one item, is not enough and the 1st in the list would need to be added. This is not clear to a user.

After adding 

```
    <dependency>
      <groupId>org.kie.kogito</groupId>
      <artifactId>kogito-quarkus</artifactId>
    </dependency>
```

The startup complains

```
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor#generateModel threw an exception: org.kie.kogito.quarkus.common.deployment.MissingRestCapabilityException: No REST capability detected! 
Add the RestEasy (quarkus-resteasy) and RestEasy Jackson (quarkus-resteasy-jackson) extensions if you want Kogito to generate REST endpoints automatically. 
RestEasy Reactive extension is currently not supported: see https://issues.redhat.com/browse/KOGITO-6131 
You may also disable automated REST generation by setting `kogito.generate.rest = false`. 
You may also override this notice by setting `kogito.generate.rest = true` 
```

Which means that those 2 dependencies should be added by default as well.

**Comment by maxandersen** [comment]
yeah, in this case we should be able to add required ones if user haven't selected them.

I'm going to move this to quarkus as same happens with mvn/gradle/quarkus cli.

@aloubyansky this feels like something we could handle at code gen and add missing required extensions ?
